### PR TITLE
Fix Repo Creation flow

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 1]
```diff
@@ -327,6 +327,7 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
             LOGGER);
         repository.update(request);
       }
+      repository.setDeleted(true);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:329]: we are again holding the transaction open till the uac operation is completed. 
can we close the transaction post the saveOrUpdate on next line.

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 2]
```diff
@@ -342,9 +343,17 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
...
+        try {
+          createRoleBindingsForRepository(request, userInfo, repository);
+          repository.setDeleted(false);
+          session.update(repository);
+        } catch (Exception ex) {
+          session.delete(repository);
+          throw ex;
+        } finally {
+          session.getTransaction().commit();
+        }
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: we should rely on the deletion cron to clean up.

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 3]
```diff
@@ -342,9 +343,17 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
         saveBranch(
             session, commitEntity.getCommit_hash(), ModelDBConstants.MASTER_BRANCH, repository);
       }
-      session.getTransaction().commit();
       if (create) {
-        createRoleBindingsForRepository(request, userInfo, repository);
+        try {
+          createRoleBindingsForRepository(request, userInfo, repository);
+          repository.setDeleted(false);
+          session.update(repository);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: we should wrap this is a separate transaction

**Review by ravishetye — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:329]: Fixed

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java (hunk 3 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: Removed from here and leave it for cron job

**Review by ravishetye — APPROVED** [review]

**Pull request merged**